release/models/interfaces/openconfig-if-rates.yang

```diff
@@ -0,0 +1,107 @@
+module openconfig-if-rates {
+
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/interfaces/rates";
+
+  prefix "oc-if-rates";
+
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-interfaces { prefix "oc-if"; }
+
+  organization
+    "OpenConfig working group";
+
+  contact
+    "www.openconfig.net";
+
+  description
+    "This module adds configuration and operational state for interface rates.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision 2024-04-02 {
+    description
+      "Augment of /interfaces/interface to include interface rates.";
+    reference
+      "0.1.0";
+  }
+
+  grouping interface-rates-config {
+    description
+      "Grouping of interface rates related configuration";
+
+    leaf load-interval {
+      type uint16 {
+        range "0..600";
```

Comment:
While `600` is the common max. value in the existing implementations, I don't think it should be used in the OC yang as the maximum.

Also, is `0` a valid value? If so, please add the meaning of "0" to the description.

Reply:
@hyun-arista I suggest using range 1...600

@LimeHat if you can show a evidence of a range longer than 600, please provide it and we can use it.  I couldn't find a SRLinux reference?  

If not, 600 seems good enough for now.  I would expect increasing the range limit in the future should be non-breaking as the model becomes more permissive.

Repeating and adding references here:
*  [Cisco IOSXR](https://www.cisco.com/c/en/us/td/docs/routers/asr9000/software/system_monitoring/command/reference/b-sysmon-cr-asr9k/b-sysmon-cr-asr9k_chapter_0111.html#wp1787771667) supports 600 as the max in 30 second increments.  
* [JunOS](https://www.juniper.net/documentation/us/en/software/junos/interfaces-security-devices/cli-reference/topics/ref/statement/load-interval-edit-interfaces.html) is at 180 max in 10 second increments.  
* [Arista EOS](https://www.arista.com/en/um-eos/eos-ethernet-ports) does not include details on the values supported.

Reply:
@dplore I don't have a reference for an implementation that supports >600s as of today, but what is the purpose of the limit?

E.g. there's no implementation that supports 60k MTU or `max(uint32)` is-is hello-interval, and we don't limit those values.

Reply:
It's true, OC is not very consistent about using ranges.  Having a smaller surface area to support and/or test can be useful.  IMO, it's feasible to model range constraints when there are one or more of:
- an externally defined specification 
- 2 or more implementation references
- rough consensus

I'll bring this up on the OC operators call Apr 9, 2024 and see if the operator group has enough of an opinion to say this should be included or excluded.

Reply:
Thank you, Darren. 
Indeed, it makes a lot of sense in the first case (an externally defined spec) since all other values are invalid/wrong.

I see two issues with enforcing an arbitrary limits for values that are not limited by the standards:
1. any implementer that is willing to go above the limit is forced to make an upstream model change first, wait until it is accepted (if it is accepted), and wait for the next OC release, and only then it can be properly implemented.
2. it is hard to model these limits in a consistent way. 

Could you elaborate on the point about the surface area? If the OC doesn't cap this at 600, it doesn't mean that every implementation should support higher values. In fact, even with 600, there will be differences between the implementations (as the JunOS example shows)

Reply:
we discussed at todays operators meeting - I don't have any objection to a restriction that covers a sensible operational range.  Suggest we use a range 1..600 for now and as Darren suggests, we could - given an operational use case in future - extend the range without it being a breaking change.

Reply:
Can we have some general guidance on when the ranges should be used? I'd suggest to update the OC style guide.

Otherwise, it seems pretty arbitrary (when the values are not restricted by a standard), and there's no clarity on how the "sensible operational range" is defined and in what cases. Why 600 is sensible, and 700 is not?

I'm not too concerned about the specific use case, but I do have a general concern about the consistency and the implications of imposing arbitrary restrictions.

Reply:
I agree that in this specific case, specifying a range is not much of a concern.  There are not obvious conflicts with this range.  Operationally it's not useful to average the counters for an interface over a very long period of time.    How long? Well we've said no need to support a value greater than 10 minutes. 

As far as a general rule for all of OC, I am not sure it is feasible?  If anyone has a proposal it would be a good pull request to review!

Reply:
>As far as a general rule for all of OC, I am not sure it is feasible?

It's better to be feasible because this PR sets an interesting precedent.  
My proposal would be, "don't use ranges when there's no corresponding standard that limits the values", which is in line with the current usage I think. 

Also, is changing the range in the future can be considered non-breaking? For example, if we extend this range to 900 and there's a system (e.g. an NMS) that does a strict validation and relies on the previous range.